spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreTemplate.java

```diff
@@ -400,34 +411,56 @@ private void doIfTransaction(Context ctx, Consumer<ReactiveFirestoreResourceHold
 	}
 
 	private <T> Write createUpdateWrite(T entity) {
-		String documentResourceName = buildResourceName(entity);
-		Document document = getClassMapper().entityToDocument(entity, documentResourceName);
-		return Write.newBuilder()
-				.setUpdate(document)
-				.build();
+		boolean needsAutoId = getIdValue(entity) == null;
+		String resourceName = buildResourceName(entity);
+		Document document = getClassMapper().entityToDocument(entity, resourceName);
+		Builder builder = Write.newBuilder().setUpdate(document);
+		if (needsAutoId) {
```

Comment:
I don't really see the point of having this `needsAutoId` local variable, since it's only used once.
I would just put `getIdValue(entity) == null` directly in the `if`.